### ISSUE TITLE
[Feat]: Add option for disabling "Show Local" while saving

### ISSUE THREAD

**Issue by avsthiago-kbc** [coder/code-server#5118]
## What is your suggestion?
Reuse the option `--disable-file-downloads` recently implemented on #5055 to remove the `Show Local` button when using the `Save As` option. This option allows the users to save the current file on their local machine.

Example:
<img width="650" alt="Screenshot 2022-04-19 at 14 53 51" src="https://user-images.githubusercontent.com/60617140/164008630-787c9cbd-9dd4-4b18-998e-3a8a9e39e44f.png">

## Why do you want this feature?
One could argue that having the option to save the file locally makes the download as accessible as having the `Download` button when right clicking the file name. As mentioned in the comments on the issue #1386, some industries have concerns about downloading files from the IDE.

## Are there any workarounds to get this functionality today?
Not that I'm aware of.

## How this feature could be implemented?
We could extend the code implemented on #5055. The value of the option `--disable-file-downloads` could be used within a condition that will either add the "Show Local" button or not. As I understand, the button is added on [simpleFileDialog.ts L279](https://github.com/microsoft/vscode/blob/12f99e06d005e5d871a1b1b264719e12ae107353/src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts#L279). _Disclaimer, I'm not familiar with the code base, neither TypeScript, so there might be more places we will need patch._

## Are you interested in submitting a PR for this?
Yes, but I might need some guidance setting up the environment.

**Comment by jsjoeio** [comment]
> This option allows the users to save the current file on their local machine.

Interesting! So this gives you access to the files on the local machine? I don't think we realized that. This is an interesting feature request though.

**Comment by avsthiago** [comment]
Yes, it opens an explorer/finder popup asking where to save the file locally. After that the file gets downloaded.

This button `Show Local` also appears when you are opening a file via `File -> Open File...`. If we decide to disable the button for the download in the [simpleFileDialog.ts](https://github.com/microsoft/vscode/blob/12f99e06d005e5d871a1b1b264719e12ae107353/src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts#L279) it might also impact the upload. We have to be aware of this side effect.

**Comment by avsthiago-kbc** [comment]
Yes, it opens an explorer/finder popup asking where to save the file locally. After that the file gets downloaded.

This button Show Local also appears when you are opening a file via File -> Open File.... If we decide to disable the button for the download in the [simpleFileDialog.ts](https://github.com/microsoft/vscode/blob/12f99e06d005e5d871a1b1b264719e12ae107353/src/vs/workbench/services/dialogs/browser/simpleFileDialog.ts#L279) it might also impact the upload. We have to be aware of this side effect.

**Comment by paulchill** [comment]
yeah it's a bit pointless removing file download but having this option, i would say --disable-file-downloads and --disable-show-local option should go hand in hand - also there is security issues with uploads too the way there are issues / code / ip leaks with downloads..

**Comment by jeffmax** [comment]
I've traced through the code and I believe the place to address this is within this function:

[addFileSchemaIfNeeded function in fileDialogService.ts](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/services/dialogs/browser/fileDialogService.ts#L44)

Specifically, my understanding is that if the `Schema.File` value ends up in `availableFileSystems` list then the "Show Local" button will be made available. I could be misunderstanding the intent behind these schemas, but in my testing, this is the behavior.

To @avsthiago-kbc comment above, this function is used in several places and would require passing a context variable to clarify whether the context is `save' or 'open'.

I have pulled together some working code for this, but one of my biggest confusion points is around the appropriate way to propagate the disable-file-downloads option down to this location in the code. Initially I was trying to follow the way this was done in the patch that introduced the flag, but it was making use of ContextKeys and using those in if-then statements didn't seem correct. I ended up using the environmentService variable that was already on abstractFileDialogService, but I had to change it from IWorkbenchEnvironmentService to IBrowserWorkbenchEnvironmentService. I noticed that this same change was made in src/vs/workbench/browser/contextkeys.ts in the PR to add the [disable-downloads flag](https://github.com/coder/code-server/pull/5055/files#diff-e6e04da7b0aaf582b2904145e107b59c0b2221bb8af7fa4f5a75789c50e357f3R103) but I don't fully understand the ramifications.

**Comment by code-asher** [comment]
In code-server that will always be an `IBrowserWorkbenchEnvironmentService` so there should be no problem.  In desktop VS Code it would be an `INativeWorkbenchEnvironmentService`.  So if we wanted to upstream this change eventually we would have to change the way we do it, but for now I think assuming it is the browser service is fine.

**Comment by jeffmax** [comment]
I just created a PR for this. I initially took the approach that I discussed in my above comment to control the contents of `availableFileSystems`, however I decided this was more confusing and modified code that I lot of actions pass through. Ultimately I modified the function that @avsthiago-kbc first mentioned as the intent was more clear and the appropriate context was available as to not disable "Show Local" for opening files (as a concern was raised above).
